pkg/migrations/sql/idempotent/base.sql

```diff
@@ -610,6 +610,30 @@ $$
 LANGUAGE SQL STABLE PARALLEL SAFE;
 GRANT EXECUTE ON FUNCTION SCHEMA_PROM.label_key_position(text, text) to prom_reader;
 
+-- drop_metric deletes a metric and related series hypertable from the database along with the related series, views and unreferenced labels.
+CREATE OR REPLACE FUNCTION SCHEMA_PROM.drop_metric(metric_name_to_be_dropped text) RETURNS VOID
+AS
+$$
+    DECLARE
+        hypertable_name TEXT;
+        deletable_metric_id INTEGER;
+    BEGIN
+        SELECT table_name, id INTO hypertable_name, deletable_metric_id FROM SCHEMA_CATALOG.metric WHERE metric_name=metric_name_to_be_dropped;
```

Comment:
we need to make sure no promscale instances are running when this function executes. This needs to happen with the following logic:
1) SELECT * FROM pg_try_advisory_lock(SCHEMA_LOCK_ID); 
2) If 1 succeeds good, we're done. Else, print a notice that to run this function you need to stop running promscale connectors and then run: `PERFORM pg_advisory_lock(SCHEMA_LOCK_ID)` <-- that's the blocking version of he command in step 1. 

note that SCHEMA_LOCK_ID is not yet passed down from the go to the sql. You will have to do that. The variable in go is in runner.go and called `schemaLockId`. The injection happens in `replaceSchemaNames` in migrate.go.